### PR TITLE
bug(acumen_order_error, concerns/*): update error handling

### PR DESCRIPTION
**Pull request by jhaltercmu2020**
update error handling

### CODE REVIEW

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb [hunk 1]
```diff
@@ -41,6 +41,7 @@ def process_invoice_detail_response(raw_data)
         results << mapped_item
       rescue => error
         issue_error(AcumenOrderError.new(
+          500,
           'process_invoice_detail_response',
           'Failed while processing Invoice_Detail record',
           invoice_details,
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: THis probably shouldn't be the entire payload. Just the ID

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb [hunk 2]
```diff
@@ -90,6 +91,7 @@ def map_line_item_data(invoices, line_item_data)
         end
       rescue => error
         issue_error(AcumenOrderError.new(
+          500,
           'map_line_item_data',
           'Failed to map line item data for invoice',
           { invoice: invoice, line_items: items },
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: This should just be the IDs

File: lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb [hunk 3]
```diff
@@ -48,6 +48,7 @@ def process_invoice_response(raw_data)
 
       rescue => error
         issue_error(AcumenOrderError.new(
+          500,
           'process_invoice_response',
           'Failed to load invoice records',
           { raw_data: i },
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_query_concern.rb]: This should just be the id of the record that failed

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

**Comment by jhaltercmu2020** [comment]
@mrogers-5s Ok I think I resolved these things but let me know if something still looks off.

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb [hunk 4]
```diff
@@ -41,9 +41,10 @@ def process_invoice_detail_response(raw_data)
         results << mapped_item
       rescue => error
         issue_error(AcumenOrderError.new(
+          error[:status],
           'process_invoice_detail_response',
           'Failed while processing Invoice_Detail record',
-          invoice_details,
+          { invoice_details_id: invoice_details[:id] },
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: invoice_details is not a hash here. It's XML data

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb [hunk 5]
```diff
@@ -41,9 +41,10 @@ def process_invoice_detail_response(raw_data)
         results << mapped_item
       rescue => error
         issue_error(AcumenOrderError.new(
+          error[:status],
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: This won't work.

This assumes that `error` is a hash -- it's not. It's an Error object. The correct syntax would be `error.status`
However, that is also incorrect as it assumes that `error` will always have a field called `status`, and we don't know that.

If we're going to set a `status` on the `AcumenAgentError`, then we need to somehow get a relevant status -- 404, 422, 503, etc.

If we don't have enough to identify which status code to use, then just use `500`.

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb (hunk 4 above)
Comment (jhaltercmu2020) [reply lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: @mrogers-5s Made some more updates here let me know what you think.

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb (hunk 5 above)
Comment (jhaltercmu2020) [reply lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: @mrogers-5s I think I fixed this as well, let me know what you think.

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb [hunk 6]
```diff
@@ -41,9 +41,10 @@ def process_invoice_detail_response(raw_data)
         results << mapped_item
       rescue => error
         issue_error(AcumenOrderError.new(
+          500,
           'process_invoice_detail_response',
           'Failed while processing Invoice_Detail record',
-          invoice_details,
+          { invoice_details_id: get_field_value(i, 'Invoice_Detail.id') },
```
Comment (mrogers-5s) [inline lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: @Jhalter5Stones `i` doesn't exist. I think you mean `invoice_details`

Also, the `Invoice_Detail` field is not `id` Look at the Acumen query to get the correct field name

**Review by mrogers-5s — CHANGES_REQUESTED** [review]

**Review by Ken-Michalak — APPROVED** [review]

File: lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb (hunk 6 above)
Comment (jhaltercmu2020) [reply lib/huginn_acumen_order_agent/concerns/invoice_detail_query_concern.rb]: @mrogers-5s Fixed.

**Review by mrogers-5s — APPROVED** [review]

**Pull request merged**